### DISCUSSION TITLE
Fetching all friendships at version 5

### DISCUSSION THREAD

**Discussion by emreond** [sbertix/Swiftagram#231] [Q&A] [answered]
Hello,

I was fetching all friendships with below code in version ~4, as I see in version ~5 friendship values return by one by. Is there anyway to get whole in once? (Without storing manually in an array etc..)
New Way:

```
        Endpoint.user(secret.identifier)
            .friendship
            .unlock(with: secret)
            .session(.instagram)
            .sink { subscriber in
                debugPrint(subscriber)
                switch subscriber {
                case .finished:
                    break
                case .failure(let error):
                    break
                }
            } receiveValue: { friendShip in
                debugPrint(friendShip)
            }.store(in: &bin)
```

Old Way:
```
        Endpoint.Friendship.summary(for: idsArray)
            .unlocking(with: secret)
            .task { (result) in
                if let friendsShips = try? result.get().friendships {
                   
                } else if case .failure(let error) = result {
                    debugPrint(error)
                }
            }.resume()
```

**Comment by sbertix** [comment, marked as answer]
Hey @emreond, 

Sure.

```swift
Endpoint.users(idsArray)
   .friendships
   .unlock(with: secret)
   .session(.instagram)
   […]
```

[Here's the migration guide](https://github.com/sbertix/Swiftagram/blob/main/docs/MIGRATION_GUIDE.md) for all `Endpoint`s from `4.*` to `5.*`.

**Reply by emreond** [reply]
Hey @sbertix , thank you for your super-duper fast response. May I ask what is [...] because as I see it only supports sink with one returned value.

**Reply by sbertix** [reply]
I just added `[…]` as not to write the entire implementation, as the thing that mattered was the actual `Endpoint` hahaha
You would still need to `sink` and `store` it 💪

**Reply by emreond** [reply]
So this endpoint returns values one by one?  If yes, how can I match a friendship object with an identifier? On old implementation I was using dictionary because It was also returning Ids.

**Reply by sbertix** [reply]
This does exactly what the old one did 😊 
It returns a `Friendship.Dictionary` and calling `friendships` on it just gets the dictionary you're looking for.
